tfx/orchestration/experimental/kubernetes/kubernetes_remote_runner.py

```diff
@@ -0,0 +1,194 @@
+  pod_label = kube_utils.sanitize_pod_name(job_name)
+  container_name = 'pipeline-orchestrator'
+  job = kube_utils.make_job_object(
+      name=job_name,
+      container_image=tfx_image,
+      command=_ORCHESTRATOR_COMMAND + arguments,
+      container_name=container_name,
+      pod_labels={
+          'job-name': pod_label,
+      },
+  )
+  try:
+    batch_api.create_namespaced_job(
+        'default', job, pretty=True)
+  except client.rest.ApiException as e:
+    raise RuntimeError('Failed to submit job! \nReason: %s\nBody: %s' %
+                       (e.reason, e.body))
+
+  # Wait for pod to start.
+  orchestrator_pods = []
+  core_api = kube_utils.make_core_v1_api()
+  start_time = datetime.datetime.utcnow()
+
+  # Wait for the kubernetes job to launch a pod.
+  while not orchestrator_pods and (
+      datetime.datetime.utcnow() - start_time).seconds < JOB_CREATION_TIMEOUT:
+    try:
+      orchestrator_pods = core_api.list_namespaced_pod(
+          namespace='default',
+          label_selector='job-name={}'.format(pod_label)
+      ).items
+    except client.rest.ApiException as e:
+      if e.status != 404:
+        raise RuntimeError('Unknown error! \nReason: %s\nBody: %s' %
+                           (e.reason, e.body))
+    time.sleep(1)
+
+  # Transient orchestrator should only have 1 pod.
+  if len(orchestrator_pods) != 1:
+    raise RuntimeError('Expected 1 pod launched by Kubernetes job, found %d' %
+                       len(orchestrator_pods))
+  orchestrator_pod = orchestrator_pods.pop()
+  pod_name = orchestrator_pod.metadata.name
+
+  absl.logging.info('Waiting for pod "default:%s" to start.', pod_name)
+  kube_utils.wait_pod(
+      core_api,
+      pod_name,
+      'default',
+      exit_condition_lambda=kube_utils.pod_is_not_pending,
+      condition_description='non-pending status')
+
+  # Stream logs from orchestrator pod.
+  absl.logging.info('Start log streaming for pod "default:%s".', pod_name)
+  try:
+    logs = core_api.read_namespaced_pod_log(
+        name=pod_name,
+        namespace='default',
+        container=container_name,
+        follow=True,
+        _preload_content=False).stream()
+  except client.rest.ApiException as e:
+    raise RuntimeError(
+        'Failed to stream the logs from the pod!\nReason: %s\nBody: %s' %
+        (e.reason, e.body))
+
+  for log in logs:
+    absl.logging.info(log.decode().rstrip('\n'))
+
+  resp = kube_utils.wait_pod(
+      core_api,
+      pod_name,
+      'default',
+      exit_condition_lambda=kube_utils.pod_is_done,
+      condition_description='done state',
+      exponential_backoff=True)
+
+  if resp.status.phase == kube_utils.PodPhase.FAILED.value:
+    raise RuntimeError('Pod "default:%s" failed with status "%s".' %
+                       (pod_name, resp.status))
+
+
+def _serialize_pipeline(pipeline: tfx_pipeline.Pipeline) -> Text:
+  """Serializes a TFX pipeline.
+
+  To be replaced with the "portable core" of the unified TFX orchestrator:
```

Comment:
"To be replaced with the the TFX Intermediate Representation: https://github.com/tensorflow/community/pull/271 ..."

Reply:
Fixed.